self-api/src/main/java/com/selfxdsd/api/Contract.java

```diff
@@ -101,4 +110,74 @@ private Roles(){ }
          */
         public static final String PO = "PO";
     }
+
+    /**
+     * Contract's compose id.
+     */
+    class Id {
+
+        /**
+         * Full name of the Repo represented by the Project.
+         * @checkstyle VisibilityModifier (25 lines)
+         */
+        public final String repoFullName;
+        /**
+         * Contributor's username.
+         */
+        public final String contributorUsername;
+        /**
+         * Contributor/Project's provider.
+         */
+        public final String provider;
+        /**
+         * Contributor's role.
+         */
+        public final String role;
```

Comment:
@criske Please make these private. We should never have public attributes, no matter what.

Reply:
Why? As long the fields are final primitives/immutable is perfectly fine. 
I'll change it but adds pointless verbosity, imho. :D

Reply:
@criske Just out of principle. I cannot understand why Java allows public fields in the first place...